### PR TITLE
Revert  "Move parameters to shared memory commit #2487"

### PR DESCRIPTION
**Pull request by umar456**
Revert the Move parameters to shared memory commit #2487 and reduce the threshold to for max parameter size.

### CODE REVIEW

**Comment by 9prady9** [comment]
@umar456 Can you please fix the conflict, I think this just happened after my last two PR merges.

File: test/jit.cpp [hunk 1]
```diff
@@ -124,10 +125,15 @@ TEST(JIT, CPP_Multi_linear) {
     x.host(&hx[0]);
     y.host(&hy[0]);
```
Comment (mark-poscablo) [inline test/jit.cpp]: Looks like you don't need the `hx` and `hy` vectors anymore

File: src/backend/cpu/Array.cpp [hunk 2]
```diff
@@ -224,29 +250,7 @@ Array<T> createNodeArray(const dim4 &dims, Node_ptr node) {
...
-                               [=](const size_t prev, const Node &n) {
-                                   // getBytes returns the size of the data
-                                   // Array. Sub arrays will be represented by
-                                   // their parent size.
-                                   return prev + n.getBytes();
-                               });
-
-                if (2 * bytes > lock_bytes) { out.eval(); }
-            }
+            if (!passesJitHeuristics<T>(node.get())) { out.eval(); }
```
Comment (9prady9) [inline src/backend/cpu/Array.cpp]: Is there a specific reason, this if block isn't condensed further ?

File: src/backend/cuda/Array.hpp [hunk 3]
```diff
@@ -36,6 +36,10 @@ void evalNodes(std::vector<Param<T>> &out, std::vector<common::Node *> nodes);
 template<typename T>
 void evalMultiple(std::vector<Array<T> *> arrays);
 
+template<typename T, typename Ti, typename FUNC>
+Array<T> createNodeArray(const dim4 &odims, FUNC createNode,
+                         const Array<Ti> *children, int num_children);
+
```
Comment (9prady9) [inline src/backend/cuda/Array.hpp]: As per our slack discussion, this is just a left over. Just leaving a comment as a reminder to remove this.

**Pull request merged**